go/pkg/channel/block.go

```diff
@@ -63,24 +64,81 @@ func getSignedProposal(channelID, ccName, funcName string, signer *identity.Cryp
 
 	c, err := signer.Serialize()
 	if err != nil {
-		return nil, errors.WithMessage(err, "signer serialize")
+		return nil, fmt.Errorf("signer serialize: %v", err)
 	}
 	prop, _, err := protoutil.CreateProposalFromCIS(cb.HeaderType_ENDORSER_TRANSACTION, channelID, spec, c)
 	if err != nil {
-		return nil, errors.WithMessage(err, "create proposal")
+		return nil, fmt.Errorf("create proposal: %v", err)
 	}
 
 	var signedProp *pb.SignedProposal
 	signedProp, err = protoutil.GetSignedProposal(prop, signer)
 	if err != nil {
-		return nil, errors.WithMessage(err, "get signe proposal")
+		return nil, fmt.Errorf("get signe proposal: %v", err)
 	}
 
 	var proposalResp *pb.ProposalResponse
 	proposalResp, err = connection.ProcessProposal(context.Background(), signedProp)
 	if err != nil {
-		return nil, errors.WithMessage(err, "process proposal")
+		return nil, fmt.Errorf("process proposal: %v", err)
 	}
 
 	return proposalResp, nil
 }
+
+func MakeChannelUpdateTransaction(channelID string, signer *identity.CryptoImpl, configUpdate *cb.Envelope) (*cb.Envelope, error) {
+
+	payload, err := protoutil.UnmarshalPayload(configUpdate.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("bad payload : %v", err)
+	}
+
+	if payload.Header == nil || payload.Header.ChannelHeader == nil {
+		return nil, fmt.Errorf("bad header")
+	}
+
+	ch, err := protoutil.UnmarshalChannelHeader(payload.Header.ChannelHeader)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshall channel header : %v", err)
+	}
+
+	if ch.Type != int32(cb.HeaderType_CONFIG_UPDATE) {
+		return nil, fmt.Errorf("bad type")
+	}
+
+	if ch.ChannelId == "" {
+		return nil, fmt.Errorf("empty channel id")
+	}
+
+	if channelID == "" {
+		channelID = ch.ChannelId
+	}
+
+	if ch.ChannelId != channelID {
+		return nil, fmt.Errorf("mismatched channel ID %s != %s", ch.ChannelId, channelID)
+	}
+
+	newConfigUpdateEnv := &cb.ConfigUpdateEnvelope{}
+	if err := proto.Unmarshal(payload.Data, newConfigUpdateEnv); err != nil {
+		return nil, fmt.Errorf("bad config update env: %v", err)
+	}
+
+	if signer != nil {
+		sigHeader, err := protoutil.NewSignatureHeader(signer)
+		if err != nil {
+			return nil, fmt.Errorf("creating signature header error: %v", err)
+		}
+
+		signature := &cb.ConfigSignature{
+			SignatureHeader: protoutil.MarshalOrPanic(sigHeader),
+		}
+
+		signature.Signature, err = signer.Sign(util.ConcatenateBytes(signature.SignatureHeader, newConfigUpdateEnv.ConfigUpdate))
```

Comment:
I suggest writing your own concatenation implementation in a utility. This should be a pretty efficient implementation (using Go 1.19 generics to allow it to work with slices of any type):

```go
func concat[T any](slices ...[]T) []T {
	size := 0
	for _, slice := range slices {
		size += len(slice)
	}

	result := make([]T, size)
	i := 0
	for _, slice := range slices {
		copy(result[i:], slice)
		i += len(slice)
	}

	return result
}
```